### PR TITLE
Split clibs out and update to XZ 5.6.3

### PR DESCRIPTION
**Pull request by hasufell**
@dfordivam

### CODE REVIEW

File: xz/xz.cabal [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    relies on the <http://tukaani.org/xz/ liblzma C library>.
+
+extra-doc-files:
+  Changelog.md
+
+source-repository head
+  type:     git
+  location: https://github.com/hasufell/lzma-static.git
+
+flag system-xz
```
Comment (dfordivam) [inline xz/xz.cabal:33]: Is it better to not specify "default" here?

Does cabal picks a value as default or the build fails?

File: xz/xz.cabal [hunk 2]
```diff
@@ -0,0 +1,77 @@
+cabal-version:       2.2
+name:                xz
+version:             5.2.5.5
```
Comment (dfordivam) [inline xz/xz.cabal]: Bump this as well?

File: xz/xz.cabal [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    streams of data represented as lazy @ByteString@s. A
+    monadic incremental interface is provided as well. This package
+    relies on the <http://tukaani.org/xz/ liblzma C library>.
+
+extra-doc-files:
+  Changelog.md
+
+source-repository head
+  type:     git
+  location: https://github.com/hasufell/lzma-static.git
```
Comment (dfordivam) [inline xz/xz.cabal:30]: add `subdir:` in both cabal..

I don't see any reason to change the repo name. so the location could remain same

File: xz/xz.cabal (hunk 1 above)
Comment (hasufell) [reply xz/xz.cabal:33]: The default value of `default` is `True` 😅

https://cabal.readthedocs.io/en/stable/cabal-package-description-file.html#configuration-flags

Comment (hasufell) [reply xz/xz.cabal:33]: The default value of `default` is `True` 😅

https://cabal.readthedocs.io/en/stable/cabal-package-description-file.html#configuration-flags

**Pull request merged**